### ISSUE TITLE
Think about using a monorepo for the different packages?

### ISSUE THREAD

**Issue by emmenko** [glennsl/bs-typed-glamor#4]
Sorry for throwing all those questions at you 😅

As I did a bugfix yesterday, I had to create two PRs in two different repos. Personally, that's a bit annoying.

Would you consider instead having a single repo with all the different packages in one place? It makes contributing and managing released much much easier 🙏

Of course it's up to you to decide what you prefer 😇

**Comment by glennsl** [comment]
It used to be a monorepo (it still exists actually, see https://github.com/glennsl/bs-typed-css), but the workflow for that was even worse. While easier to PR to, it's much more cumbersome to set up for local development. If it hadn't been a monorepo, you would have been able to use this several months ago. The reason you now can is because I went away from it.

Feel free to play around with the monorepo to see if you can get it to work though. I'll definitely consider a proposal, but I'm not very hopeful :)

**Comment by emmenko** [comment]
Ah I didn't know. What were the problems you encountered?

I'll try to have a look then to see if I find some problems or not. I'll get back to you then.

**Comment by glennsl** [comment]
Hooking the local dependencies up to each other I think. Can't remember exactly why that was a problem though, but I ended up using local file path in package.json, which of course doesn't work if it's not local.

**Comment by emmenko** [comment]
Quick question: in case I try to set up a monorepo, in which repo would you like to have it?

**Comment by glennsl** [comment]
The old monorepo is fine. It's not too far behind and should be quick to get up to date.

**Comment by emmenko** [comment]
Alright, I've set up the monorepo to be able to manage the packages together (using lerna).

https://github.com/emmenko/bs-typed-css/tree/nm-monorepo-setup

> I can't open a PR to the upstream repo because it's marked as "archived". If you "re-open" it I can open the PR there.

The packages are also up-to-date with the repositories (core and glamor). I updated all deps to the latest versions (e.g. reason-react, bs-platform, etc).

For consistency I named the packages with the scope `@glennsl`

* `@glennsl/bs-typed-css-core`
* `@glennsl/bs-typed-glamor`
* `@glennsl/bs-styled-react`

I saw you were using the licence `LGPL-3.0`, so I kept that.

Other than that, this should be good to go. Let me know if you need more information.

**Comment by glennsl** [comment]
Hey. Thanks for doing this. I've tried it out and found a few issues, mostly minor but would be nice if it would be possible to resolve them:

* No start script despite the readme saying there is. `lerna run start` also doesn't seem to work properly.
* Can't consume packages directly from github, they have to be published to npm
* `lerna publish` rather than [`np`](https://github.com/sindresorhus/np). Not sure how they compare yet, but I really like `np`

Other than that it seems good, but I have a bit of a nagging feeling that there's a problem with lerna that I've forgotten since I have considered it in the past. It's possible it was because of the postinstall issue with yarn though.

I've unarchived the bs-typed-css repo. if you submit a PR there I can review and comment on the changes in more detail.
